tests/frontend/test_engine.py

```diff
@@ -166,209 +174,304 @@ def inspect():
         assert inspect() == expected2
 
 
-class TestStarshipEngine:
-    """
-    Tests various methods on the remote engine StarshipEngine.
+@pytest.fixture
+def connection():
```

Comment:
All API specific tests I would move into their own folder (i.e., `tests/api/test_connection.py`, `tests/api/test_job.py`)

Reply:
It might even make sense to move the API-specific starship tests here as well.

Reply:
Sounds good, I will do so!

Reply:
All of the new tests have been moved under a new `api` directory.